projects/packages/my-jetpack/_inc/components/my-jetpack-tab-panel/index.tsx

```diff
@@ -0,0 +1,65 @@
+import { TabPanel } from '@wordpress/components';
+import { __ } from '@wordpress/i18n';
+import { useCallback } from 'react';
+import { useNavigate, useParams } from 'react-router-dom';
+import { HelpTab } from './help-tab';
+import { OverviewTab } from './overview-tab';
+import { ProductsTab } from './products-tab';
+import styles from './styles.module.scss';
+
+const tabComponentMap = {
+	overview: OverviewTab,
+	products: ProductsTab,
+	help: HelpTab,
+};
+
+/**
+ * My Jetpack Tab panel component.
+ *
+ * @return The rendered component.
+ */
+export function MyJetpackTabPanel() {
+	const params = useParams();
+	const navigate = useNavigate();
+
+	const onTabSelect = useCallback(
+		( tabName: string ) => {
+			if ( tabName !== params.tab ) {
+				navigate( `/${ tabName }` );
+			}
+		},
+		[ navigate, params.tab ]
+	);
+
+	const tabRenderer = useCallback( ( tab: { name: string; title: string } ) => {
+		const TabComponent = tabComponentMap[ tab.name ];
+
+		if ( ! TabComponent ) {
+			return null;
+		}
+		return <TabComponent />;
+	}, [] );
+
+	return (
+		<TabPanel
+			className={ styles[ 'tab-panel' ] }
+			initialTabName={ params.tab || 'overview' }
+			onSelect={ onTabSelect }
+			children={ tabRenderer }
+			tabs={ [
+				{
+					name: 'overview',
+					title: __( 'Overview', 'jetpack-my-jetpack' ),
+				},
+				{
+					name: 'products',
+					title: __( 'Products', 'jetpack-my-jetpack' ),
+				},
+				{
+					name: 'help',
+					title: __( 'Help', 'jetpack-my-jetpack' ),
+				},
+			] }
```

Comment:
Maybe it's better to define it as a variable outside of the component?

I also lack (type) connection between this array and `tabComponentMap`. It could be more exhaustive that options provided here should match the mapping and vice-versa.

Reply:
> Maybe it's better to define it as a variable outside of the component?

That doesn't play nice with translations as the components update but the variable outside the component never updates/re-renders.

Reply:
I have extracted it to a function instead of a variable that works well with translations.